src/bluesearch/database/article.py

```diff
@@ -32,6 +33,50 @@
 from bluesearch.database.identifiers import generate_uid
 
 
+def get_arxiv_id(path: str | Path) -> str | None:
+    """Compute arXiv ID, including version, from file path.
+
+    Parameters
+    ----------
+    path
+        The file path to an arXiv article.
+
+    Returns
+    -------
+    str or None
+        arXiv ID, if possible to compute.
+
+    Raises
+    ------
+    ValueError
+        If no valid arXiv ID could be inferred from the file path.
+
+    References
+    ----------
+    https://arxiv.org/help/arxiv_identifier
+    """
+    path = Path(path)
+
+    # New format, since 2007-04, only needs path stem:
+    # - since 2015-01 have format YYMM.NNNNN (i.e. 5 digits)
+    # - up to 2014-12 have format YYMM.NNNN (i.e. 4 digits)
+    pattern = re.compile(r"\A(\d{4}\.\d{4}\d?v\d+)\Z")
+    match = re.search(pattern, path.stem)
+    if match:
+        return f"arxiv:{match.groups()[0]}"
+
+    # Old format, up to 2007-03, needs to look at the whole path:
+    # - some_path/arxiv/<archive>/<format>/YYMM/YYMMNNNv<version>.<ext>
+    # Note: format may contain "-"
+    pattern = re.compile(r"arxiv/([\w-]+)/\w+/\d{4}/(\d{7}v\d+)\.\w+\Z")
+    match = re.search(pattern, "/".join(path.parts[-5:]))
+    if match:
+        match_groups = match.groups()
+        return f"arxiv:{match_groups[0]}/{match_groups[1]}"
```

Comment:
Here also one could write `f"arxiv:{match.group(0)}/{match_group(1)}"`

Or for more readability also
```python
cat, id_ = match.groups()
return f"arxiv:{cat}/{id_}"
```

Reply:
Fixed in ec2ad5f.